test_files/async_functions/async_functions.ts

```diff
@@ -37,9 +37,9 @@ class Container {
       const s = await 3;
       return s + this.field;
     };
-    async function asyncFunctionInMethod(param: string): Promise<string> {
+    async function asyncFunctionInMethod(this: Container, param: string): Promise<string> {
       const s = await 3;
-      return s + this.field;
+      return s + (this as any).field;
```

Comment:
I think this isn't right, and might break dis/ambguation.

I can't tell from this change if it's meant to be a workaround, or if we expect user code to change in the same way?

Reply:
Good catch. I was playing around with this and didn't remove the as any cast again. This ultimately didn't need a change I think.

Reply:
The tricky bit is that TS compiler now disallows having a function without the this type specified here, so I needed to add an explicit this param type. That means a code path in tsickle is now under-exercised (never have a nested fn that needs a this * added), but I think that's OK.